### ISSUE TITLE
`brew bundle cleanup` and Ruby code in Brewfile

### ISSUE THREAD

**Issue by michaelmoussa** [Homebrew/homebrew-bundle#479]
I'm pretty sure I'm deep into "undocumented / unsupported use case" territory here, but figured I'd raise the issue just in case there's a better way to do it or an opportunity to enhance Homebrew.

Like many Homebrew users, I have a `dotfiles` repository that I use for all my macOS shell goodies. I also have a work computer running macOS for which I use the same `dotfiles`. So far so good.

Now, I have some casks for my personal machine that I don't want to install on my work machine (e.g. personal finance software, programming languages I don't use at work), and some on my work machine that I don't want to install on my personal machine (e.g. video conferencing software, work-only VPN client, etc). I couldn't find anything Homebrew docs explaining what the best way to do this would be, so I did something along these lines:

```ruby
# Current system's Machine ID
MACHINE_ID = `ioreg -rd1 -c IOPlatformExpertDevice | grep UUID`.strip!

#  Personal & Work computers' Machine IDs, so as to restrict what gets installed where
PERSONAL = "<... some uuid ...>"
WORK = "<... some uuid ...>""

# Syntactic sugar for later use
IS_PERSONAL = (MACHINE_ID == PERSONAL)
IS_WORK = (MACHINE_ID == WORK)

if IS_PERSONAL
    cask "foo"
    cask "bar"
elsif IS_WORK
    cask "baz"
end

cask "stuff"
cask "i"
cask "want"
cask "on"
cask "both"
cask "computers"
```

This works as expected. My personal computer gets `foo` and `bar` installed, my work computer gets `baz` installed, and both of them get everything else installed.

The problem is that `brew bundle cleanup` wants to remove `baz` from my work machine and `foo` and `bar` from my personal machine because it doesn't appear to execute any of the Ruby code I've added to the `Brewfile`, so the conditions don't take effect.

**SO...**

* Is what I'm doing above perfectly fine, which would only require a fairly easy update to the `brew bundle cleanup` implementation to avoid the issue I'm having?
* Is there a better/supported/encouraged way to do what I'm trying to do above instead, which would _not_ cause issues when running `brew bundle cleanup`?

**Comment by MikeMcQuaid** [comment]
> * which would only require a fairly easy update to the `brew bundle cleanup` implementation to avoid the issue I'm having?

I'm not sure whether it'll be hard or easy but I'll review a PR. Basically if there's <10 lines required to change this (except tests) then we'd consider it.

> * Is there a better/supported/encouraged way to do what I'm trying to do above instead, which would _not_ cause issues when running `brew bundle cleanup`?

Probably use two separate Brewfiles and just accept a certain amount of duplication.

Closing but will review a PR.

**Comment by michaelmoussa** [comment]
Whoops.... turns out there was a bug in my Ruby code causing my problem (not surprising, as I'm not a Ruby developer - oops). The bug _didn't_ make it into the minimal example I re-typed for this issue, so what I shared above actually would work in real life. I actually did start working on a PR for this before realizing my mistake. Fixed the issue in my setup and now all is working as expected.

Hopefully someone else looking for a way to conditionally install formulae/casks will stumble upon this and find it useful, so at least it wouldn't have been a complete waste of time.

Thanks for the quick response anyway!
